syntax = "proto3";

package user;

service GrpcUserService {
  rpc GetEventOrganizer(GetEventOrganizerRequest) returns (GetEventOrganizerResponse);
  rpc GetSystemUser(GetSystemUserRequest) returns (GetSystemUserResponse);
  rpc GetReferralInfo(GetReferralInfoRequest) returns (GetReferralInfoResponse);
}

message GetEventOrganizerRequest {
    string ticket_type_id = 1;
}

message GetEventOrganizerResponse {
    optional string error = 1;
    string user_id = 2;
}

message GetSystemUserRequest {}

message GetSystemUserResponse {
  optional string error = 1;
  string user_id = 2;
}

message GetReferralInfoRequest {
  string user_id = 1;
}

message GetReferralInfoResponse {
  optional string error = 1;
  optional string user_id = 2;
}
